packages/vc-export/src/Presentation.ts

```diff
@@ -0,0 +1,379 @@
+}
+
+/**
+ * Signs a presentation in its JWT rendering.
+ *
+ * @param presentation The VerifiablePresentation (without proof).
+ * @param signingKey Key object required for signing.
+ * @param signingKey.secretKey The bytes of the secret key.
+ * @param signingKey.keyUri The key uri by which the public key can be looked up from a DID document.
+ * @param signingKey.type The key type. Ed25519 and ecdsa (secp256k1) are supported.
+ * @param options Additional optional configuration.
+ * @param options.validFrom Timestamp (in ms since the epoch) indicating the earliest point in time where the presentation becomes valid. Defaults to the current time.
+ * @param options.expiresIn Duration of validity of the presentation in ms. If omitted, the presentation's validity is unlimited.
+ * @param options.challenge Optional challenge provided by a verifier that can be used to prevent replay attacks.
+ * @param options.audience Identifier of the verifier to prevent unintended re-use of the presentation.
+ * @returns A signed JWT in compact representation containing a VerifiablePresentation.
+ */
+export function signJwt(
+  presentation: VerifiablePresentation,
+  signingKey: {
+    secretKey: Uint8Array
+    keyUri: DidResourceUri
+    type: supportedKeys
+  },
+  options: JwtPayloadArgs = {}
+): string {
+  const encodedPayload = toJwtPayload(presentation, options)
+
+  const { type, keyUri, secretKey } = signingKey
+  const alg = { ecdsa: 'ES256K', ed25519: 'EdDSA' }[type]
+  if (!alg)
+    throw new Error(`no signature algorithm available for key type ${type}`)
+  const jwtHeader = {
+    alg,
+    kid: keyUri,
+    type: 'JWT',
+  }
+
+  const signData = `${jwtSerialize(jwtHeader)}.${encodedPayload}`
+
+  const signature = encodeBase64url(
+    signers[type](stringToU8a(signData), secretKey)
+  )
+
+  return `${signData}.${signature}`
+}
+
+const verifiers: Record<VerificationKeyType, VerifierFunction> = {
+  ecdsa: (message, signature, publicKey) =>
+    secp256k1Verify(signature, sha256AsU8a(message), publicKey),
+  ed25519: ed25519Verify,
+  sr25519: () => {
+    throw new Error('not implemented')
+  },
+}
+
+/**
+ * Verifies a JWT rendering of a VerifiablePresentation.
+ *
+ * @param jwt The JWT in compact (string) encoding.
+ * @param options Optional configuration.
+ * @param options.audience Expected audience. Verification fails if the aud claim in the JWT is not equal to this value.
+ * @param options.challenge Expected challenge. Verification fails if the nonce claim in the JWT is not equal to this value.
+ * @returns The VerifiablePresentation (without proof), the decoded JWT payload containing all claims, and the decoded JWT header.
+ */
+export async function verifyJwt(
+  jwt: string,
+  { audience, challenge }: { audience?: string; challenge?: string }
+): Promise<{
+  presentation: VerifiablePresentation
+  payload: Record<string, AnyJson>
+  header: Record<string, AnyJson>
+}> {
+  const parts = jwt.match(
```

Comment:
You mean relying more on `did-jwt` instead of reimplementing these parts? I was debating that, to me it seemed that its not a crazy amount of code so that writing a few lines of code seemed preferable to adding another dependency. JWTs are not complicated; the only thing missing from our feature set in order to produce one is literally just base64 encoding two JSON structures and signing their concatenation.
But yeah what's your preference here? I am using `did-jwt` in the tests to demonstrate interoperability (I think just verification at this point though). Do you like that better?

Reply:
Stuff won't probably change that fast in the JWT standard, but I personally think it's always better to rely on a standard library to do what we are indeed trying to do, which is being JWT-compliant. 
Also because perhaps you know what the different steps are, but I assume that's not valid for everyone else in the team, and offloading that to the library takes off most of the complexity from our shoulders.

Reply:
I'll give that a try, see how it looks